### PR TITLE
fix(emqx_connection): handle socket activation error return

### PR DESCRIPTION
**Pull request by zmstone**
Fixes [EMQX-11431](https://emqx.atlassian.net/browse/EMQX-11431)

This fix covers two `emqx_connection` process crashes when trying to set socket options after the socket is closed already.

2023-11-17T19:25:08.417154+00:00 [error] crasher: initial call: emqx_connection:init/4, pid: <0.1339.2233>, registered_name: [], error: {{case_clause,{error,einval}},[{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,493}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,497}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,499}]},{emqx_connection,handle_recv,3,[{file,"emqx_connection.erl"},{line,455}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,250}]}]}, 

2023-11-20T02:36:35.569334+00:00 [error] crasher: initial call: emqx_connection:init/4, pid: <0.11021.5006>, registered_name: [], error: {{case_clause,{error,{options,{socket_options,{active,100}}}}},[{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,493}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,497}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,499}]},{emqx_connection,handle_recv,3,[{file,"emqx_connection.erl"},{line,455}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,250}]}]},

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 8ec3b1d</samp>

Fix a connection hang issue when rate limiter is enabled by using `handle_info` instead of `activate_socket` in `emqx_connection.erl`.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/src/emqx_connection.erl [hunk 1]
```diff
@@ -552,13 +552,13 @@ handle_msg({quic, Data, _Stream, #{len := Len}}, State) when is_binary(Data) ->
     inc_counter(incoming_bytes, Len),
     ok = emqx_metrics:inc('bytes.received', Len),
     when_bytes_in(Len, Data, State);
-handle_msg(check_cache, #state{limiter_buffer = Cache} = State) ->
-    case queue:peek(Cache) of
+handle_msg(check_limiter_buffer, #state{limiter_buffer = Buffer} = State) ->
+    case queue:peek(Buffer) of
         empty ->
-            activate_socket(State);
+            handle_info(activate_socket, State);
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_connection.erl:558]: should it follow L928?

Comment (zmstone) [reply apps/emqx/src/emqx_connection.erl:558]: this covers L928.

Comment (zmstone) [reply apps/emqx/src/emqx_connection.erl:558]: did the same as 
https://github.com/emqx/emqx/blob/d99f033d7491836f1f62b8704ecb8fcf6661d302/apps/emqx/src/emqx_connection.erl#L596

Comment (qzhuyan) [reply apps/emqx/src/emqx_connection.erl:558]: ok, I have `handle_info({sock_error, Reason}, State)` on my branch. maybe it is lagging.

**Review by qzhuyan — DISMISSED** [review]

File: apps/emqx/src/emqx_connection.erl (hunk 1 above)
Comment (qzhuyan) [reply apps/emqx/src/emqx_connection.erl:558]: https://github.com/emqx/emqx/blob/d99f033d7491836f1f62b8704ecb8fcf6661d302/apps/emqx/src/emqx_connection.erl#L928

Comment (qzhuyan) [reply apps/emqx/src/emqx_connection.erl:558]: I don't like the idea of calling handle_info when handling `check_limiter_buffer`, it confused reader.

Comment (zmstone) [reply apps/emqx/src/emqx_connection.erl:558]: I agree, it confused me too.

**Review by qzhuyan — APPROVED** [review]

**Pull request merged**